spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/JpaMetamodelEntityInformation.java

```diff
@@ -222,6 +225,29 @@ public boolean isNew(T entity) {
 		return versionAttribute.map(it -> wrapper.getPropertyValue(it.getName()) == null).orElse(true);
 	}
 
+	@Override
+	public Map<String, Object> getKeyset(Iterable<String> propertyPaths, T entity) {
+
+		// TODO: proxy business?
```

Comment:
If we can't resolve the TODO now we should leave clearer instructions what exactly the TODO is, and potentially create a ticket for it.